lib/iris/cube.py

```diff
@@ -3583,6 +3583,15 @@ def __eq__(self, other):
                     or coord_comparison["non_equal_data_dimension"]
                 )
 
+            if result:
+                anc_diff = set(self._ancillary_variables_and_dims) == set(
+                    other._ancillary_variables_and_dims
+                )
+                cm_diff = set(self._cell_measures_and_dims) == set(
```

Comment:
It might be worth "shortcutting" again, by putting both of these checks into an "if result:" block.
For choice, I think we should check cell-measures first, as being typically X/Y (geo-grid) based, they will commonly have fewer dimensions than ancillary variables -- which could often be as big as the whole main data array.
I think it is also slightly neater for all the checks to work the same way : = "test result and if not already False set it again", that is instead of grouping test results with 'and'.

Reply:
Good call.